Update important-notes.mdx (#237)
chainflip-io/chainflip-docs: pages/mainnet/validator-setup/important-notes.mdx

---

title: Security & Redundancy

> description: Important notes about the Chainflip network; how to properly secure the ChainFlip Protocol and Validators.

Reviewer: ```suggestion
description: Important notes about the Chainflip network; how to properly secure the Chainflip Protocol and Validators.
```

## Backups

It is crucial to have a backup and recovery strategy in place due to the possibility of data loss or corruption caused by hardware failures, unexpected interruptions from cloud providers, loss of access to the machine, or accidental deletion. It is also important to understand the separate, yet complimentary, concepts of providing security and redundancy to the ChainFlip Network and also to the Validator's $FLIP bond. There are absolute minimum backup requirements and some best practices to increase your chances of a successfuly recovery.

### Absolute minimum backup requirements

#### Validator's $FLIP Bond

> Backing up and securing the Validator's $FLIP bond is akin to backing up any cryptocurrency wallet. The seed-phrase that was created by [generating validator keys](/mainnet/validator-setup/keys#validator-keys) must be securely and remotely stored; ideally in a physically secure location offline. This seed-phrase will be required to recover your ChainFlip account keys and your Ethereum keys. This is the absolute minimum requirement to backing up your Validator's $FLIP bond.

Reviewer: ```suggestion
Backing up and securing the Validator's $FLIP bond is akin to backing up any cryptocurrency wallet. The seed-phrase that was created by [generating validator keys](/mainnet/validator-setup/keys#validator-keys) must be securely and remotely stored; ideally in a physically secure location offline. This seed-phrase will be required to recover your Chainflip account keys and your Ethereum keys. This is the absolute minimum requirement to backing up your Validator's $FLIP bond.
```

- The same applies to engine key shares. Losing these keys will prevent you from participating in keygen ceremonies or performing any multi-signature operations. Failing a keygen will result in penalties and a potential ban, which could lead to your validator node being removed from the authority set. In such cases, you would have to wait for the next auction to rejoin.
- If you think the keys might be compromised, we recommend that you immediately secure your FLIP funds by [binding a redeem address](/testnet/funding/advanced-account-management#redemption-execution-address-binding), [stopping your bid](/mainnet/validator-setup/funding-and-bidding) and then, when your funds are unbonded, redeeming them to your wallet before staking a new node with new keys.

#### Engine Keyshares

The keyshares use by the `chainflip-engine` are stored in a database. By default this is located at `/etc/chainflip/data.db`.

> Note it is *not* possible to recover these key shares from a seed if they are lost, so it it crucial to back up this database after each epoch.

Reviewer: ```suggestion
Note it is *not* possible to recover these key shares from a seed if they are lost, so it it crucial to back up this database, ideally at the beginning of each epoch.
```

Technically almost the same meaning, but since keys are fresh at the start of the epoch, might as well back them up right away.
